packages/taiko-client/driver/chain_syncer/blob/syncer.go

```diff
@@ -377,7 +377,7 @@ func (s *Syncer) insertNewHead(
 		}
 
 		var parentGasExcess uint64
-		if newGasTarget != parentGasTarget {
+		if newGasTarget != parentGasTarget && parentGasTarget != 0 {
```

Comment:
lets create a separate PR for this.

Reply:
This is to solve the issue of inconsistency between client and contract.